### PR TITLE
Add bridge-marker manifests

### PR DESCRIPTION
**Pull request by mmirecki**
**What this PR does / why we need it**:
This PR adds bridge-marker manifests to the linux bridge operator

**Release note**:
```release-note
The linux bridge operator will now add the bridge-marker daemon set.
```

### CODE REVIEW

File: data/linux-bridge/0004-bridge-marker-daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: bridge-marker
+  namespace: kube-system
```
Comment (phoracek) [inline data/linux-bridge/0004-bridge-marker-daemonset.yaml]: Please put it in the same namespace as the bridge CNI.

File: data/linux-bridge/0004-bridge-marker-daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+      containers:
+      - name: bridge-marker
+        image: quay.io/kubevirt/bridge-marker:latest
```
Comment (phoracek) [inline data/linux-bridge/0004-bridge-marker-daemonset.yaml]: has to be configurable, look at the other manifest

File: data/linux-bridge/0004-bridge-marker-daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        imagePullPolicy: Always
+        args:
+          - -node-name
+          - $(NODE_NAME)
+        env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+
```
Comment (phoracek) [inline data/linux-bridge/0004-bridge-marker-daemonset.yaml]: please add a newline

File: data/linux-bridge/003-bridge-marker-rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+- kind: ServiceAccount
+  name: bridge-marker
+  namespace: kube-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: bridge-marker
+  namespace: kube-system
+
```
Comment (phoracek) [inline data/linux-bridge/003-bridge-marker-rbac.yaml]: drop empty line, add new line to the end of the file

File: data/linux-bridge/0004-bridge-marker-daemonset.yaml (hunk 1 above)
Comment (mmirecki) [reply data/linux-bridge/0004-bridge-marker-daemonset.yaml]: Done

File: manifests/cluster-network-addons/0.4.0/operator.yaml [hunk 5]
```diff
@@ -131,6 +131,8 @@ spec:
           value: docker.io/nfvpe/multus:latest
         - name: LINUX_BRIDGE_IMAGE
           value: quay.io/kubevirt/cni-default-plugins:latest
+        - name: BRIDGE_MARKER_IMAGE
+          value: quay.io/kubevirt/bridge-marker
```
Comment (phoracek) [inline manifests/cluster-network-addons/0.4.0/operator.yaml]: please release bridge marker with a tag and keep tagged version here

File: manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -151,6 +151,8 @@ spec:
                     value: docker.io/nfvpe/multus:latest
                   - name: LINUX_BRIDGE_IMAGE
                     value: quay.io/kubevirt/cni-default-plugins:latest
+                  - name: BRIDGE_MARKER_IMAGE
```
Comment (phoracek) [inline manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml]: there should be no changes under 0.4.0. this directory is automatically generated when the operator is released.

File: pkg/components/components.go [hunk 7]
```diff
@@ -24,6 +25,7 @@ const (
 type AddonsImages struct {
 	Multus         string
 	LinuxBridgeCni string
+	BridgeMarker   string
```
Comment (phoracek) [inline pkg/components/components.go]: Keep this consistend with bridge CNI, use `LinuxBridgeMarker` instead

File: README.md [hunk 8]
```diff
@@ -60,6 +60,10 @@ spec:
 Additionally, container image used to deliver this plugin can be set using
 `LINUX_BRIDGE_IMAGE` environment variable in operator deployment manifest.
 
+The bridge marker image used to deliver a bridge marker detecting the availability
+of linux bridges on nodes can be set using the `BRIDGE_MARKER_IMAGE` environment
```
Comment (phoracek) [inline README.md]: LINUX_BRIDGE_MARKER_IMAGE

File: manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml (hunk 6 above)
Comment (phoracek) [reply manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml]: Please edit it in `Makefile`, `tools/manifests-templator/manifests-templator.go` and `pkg/components/components.go`. Also, please do release bridge-marker on u/s (with a tag) and create a marker image with the tag. We should not link :latest here

File: README.md (hunk 8 above)
Comment (mmirecki) [reply README.md]: done

File: data/linux-bridge/0004-bridge-marker-daemonset.yaml (hunk 2 above)
Comment (mmirecki) [reply data/linux-bridge/0004-bridge-marker-daemonset.yaml]: done

File: data/linux-bridge/003-bridge-marker-rbac.yaml (hunk 4 above)
Comment (mmirecki) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: done

File: manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml (hunk 6 above)
Comment (mmirecki) [reply manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml]: Removed the files from 0.4.0

File: pkg/components/components.go (hunk 7 above)
Comment (mmirecki) [reply pkg/components/components.go]: done

File: data/linux-bridge/003-bridge-marker-rbac.yaml (hunk 4 above)
Comment (SchSeba) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: I still see the same problem here.
Can you please take a look ?

File: manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml [hunk 9]
```diff
@@ -151,6 +151,8 @@ spec:
                     value: docker.io/nfvpe/multus:latest
                   - name: LINUX_BRIDGE_IMAGE
                     value: quay.io/kubevirt/cni-default-plugins:latest
+                  - name: LINUX_BRIDGE_MARKER_IMAGE
```
Comment (SchSeba) [inline manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml]: we should not change any manifest file.
After this PR is merge we can open a new one with a new release

File: pkg/components/components.go [hunk 10]
```diff
@@ -14,19 +14,21 @@ import (
...
 
 const (
-	MultusImageDefault         = "docker.io/nfvpe/multus:latest"
-	LinuxBridgeCniImageDefault = "quay.io/kubevirt/cni-default-plugins:latest"
-	SriovDpImageDefault        = "quay.io/booxter/sriov-device-plugin:latest"
-	SriovCniImageDefault       = "docker.io/nfvpe/sriov-cni:latest"
-	KubeMacPoolImageDefault    = "quay.io/schseba/mac-controller:latest"
+	MultusImageDefault            = "docker.io/nfvpe/multus:latest"
+	LinuxBridgeCniImageDefault    = "quay.io/kubevirt/cni-default-plugins:latest"
+	LinuxBridgeMarkerImageDefault = "quay.io/kubevirt/bridge-marker:latest"
```
Comment (phoracek) [inline pkg/components/components.go]: please release a tagged image and pin it here

Comment (mmirecki) [reply pkg/components/components.go]: Added 0.1.0

File: manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml (hunk 9 above)
Comment (mmirecki) [reply manifests/cluster-network-addons/0.4.0/cluster-network-addons-operator.0.4.0.clusterserviceversion.yaml]: removed

File: manifests/cluster-network-addons/0.4.0/operator.yaml (hunk 5 above)
Comment (mmirecki) [reply manifests/cluster-network-addons/0.4.0/operator.yaml]: added 0.1.0

File: data/linux-bridge/003-bridge-marker-rbac.yaml (hunk 4 above)
Comment (mmirecki) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: There is a newline at the end of the file. Should this be dropped?

Comment (SchSeba) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: now it looks good thanks! :)

File: data/linux-bridge/0004-bridge-marker-daemonset.yaml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    app: bridge-marker
+spec:
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: bridge-marker
+    spec:
+      serviceAccountName: bridge-marker
+      hostNetwork: true
```
Comment (phoracek) [inline data/linux-bridge/0004-bridge-marker-daemonset.yaml]: Please test this code on OpenShift. I believe it will fail due to SCC.

Comment (mmirecki) [reply data/linux-bridge/0004-bridge-marker-daemonset.yaml]: Yes, the bridge marker user needs the hostnetwork scc.

Comment (mmirecki) [reply data/linux-bridge/0004-bridge-marker-daemonset.yaml]: added a bridge-marker scc

File: data/linux-bridge/003-bridge-marker-rbac.yaml [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  name: bridge-marker
+  namespace: linux-bridge
+{{ if .EnableSCC }}
+---
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  name: bridge-marker
+allowHostNetwork: true
+allowHostPorts: true
```
Comment (phoracek) [inline data/linux-bridge/003-bridge-marker-rbac.yaml]: Why do we need this?

Comment (mmirecki) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: Yes, the pod will not be scheduled without allowHostPorts.

Comment (phoracek) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: I wonder why. Could you please share events from the DS and respective replica set? You should find them in the `kubectl describe ds bridge-marker` and `kubectl describe replicaset bridge-marker` (or something similar, not sure if it gets suffix).

Comment (phoracek) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: Or just point me to the place in manifests where we request host ports.

Comment (mmirecki) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: tried this out again. Indeed it works without allowHostPorts (why did it not previously?)
Pushed an update.

Comment (phoracek) [reply data/linux-bridge/003-bridge-marker-rbac.yaml]: Thanks :)

**Comment by SchSeba** [comment]
Thanks @mmirecki nice job!

**Pull request merged**